kamon-core/src/main/scala/kamon/trace/TracerModule.scala

```diff
@@ -122,12 +122,12 @@ private[kamon] class TracerModuleImpl(metricsExtension: MetricsModule, config: C
 
     val traceToken = token.getOrElse(newToken)
 
-    if (_settings.levelOfDetail == LevelOfDetail.MetricsOnly || !isLocal)
-      newMetricsOnlyContext(traceToken)
-    else {
-      if (!_settings.sampler.shouldTrace)
+    _settings.levelOfDetail match {
+      case LevelOfDetail.MetricsOnly ⇒
         newMetricsOnlyContext(traceToken)
-      else
+      case _ if !isLocal || !_settings.sampler.shouldTrace ⇒
+        newMetricsOnlyContext(traceToken)
+      case _ ⇒
         new TracingContext(traceName, traceToken, true, _settings.levelOfDetail, isLocal, startTimestamp, null, dispatchTracingContext)
```

Comment:
Yeap, that start call seem to be the right place and it is a fairly simple change, can you please include it? although the thread safety issue would be much more important to correct in order to merge this PR :).